operator/cmd/mesh/manifest-apply.go

```diff
@@ -153,78 +154,70 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, maArgs *manifestApplyAr
 	if err != nil {
 		return err
 	}
 
-	kubeconfig, err := manifest.InitK8SRestClient(kubeConfigPath, context)
+	restConfig, clientSet, err := manifest.InitK8SRestClient(kubeConfigPath, context)
 	if err != nil {
 		return err
 	}
-	manifests, iops, err := GenManifests(inFilenames, ysf, force, kubeconfig, l)
+	client, err := client.New(restConfig, client.Options{Scheme: scheme.Scheme})
 	if err != nil {
-		return fmt.Errorf("failed to generate manifest: %v", err)
+		return err
 	}
-	opts := &kubectlcmd.Options{
-		DryRun:      dryRun,
-		Verbose:     verbose,
-		Wait:        wait,
-		WaitTimeout: waitTimeout,
-		Kubeconfig:  kubeConfigPath,
-		Context:     context,
+	_, iops, err := GenerateConfig(inFilenames, ysf, force, restConfig, l)
+	if err != nil {
+		return err
 	}
 
-	for cn := range name.DeprecatedComponentNamesMap {
-		manifests[cn] = append(manifests[cn], fmt.Sprintf("# %s component has been deprecated.\n", cn))
+	crName := installedSpecCRPrefix
+	if iops.Revision != "" {
+		crName += "-" + iops.Revision
 	}
-
-	out, err := manifest.ApplyAll(manifests, version.OperatorBinaryVersion, iops, opts)
+	iop, err := translate.IOPStoIOP(iops, crName, iopv1alpha1.Namespace(iops))
 	if err != nil {
-		return fmt.Errorf("failed to apply manifest with kubectl client: %v", err)
+		return err
 	}
-	gotError := false
-
-	for cn := range manifests {
-		if out[cn].Err != nil {
-			cs := fmt.Sprintf("Component %s - manifest apply returned the following errors:", cn)
-			l.logAndPrintf("\n%s", cs)
-			l.logAndPrint("Error: ", out[cn].Err, "\n")
-			gotError = true
-		}
 
-		if !ignoreError(out[cn].Stderr) {
-			l.logAndPrint("Error detail:\n", out[cn].Stderr, "\n", out[cn].Stdout, "\n")
-			gotError = true
-		}
+	if err := manifest.CreateNamespace(iop.Namespace); err != nil {
+		return err
 	}
 
-	if gotError {
-		l.logAndPrint("\n\n✘ Errors were logged during apply operation. Please check component installation logs above.\n")
+	reconciler, err := helmreconciler.NewHelmReconciler(client, restConfig, iop, &helmreconciler.Options{DryRun: dryRun, Logger: l})
+	if err != nil {
+		return err
+	}
+	if err := reconciler.Reconcile(); err != nil {
+		l.LogAndPrint("\n\n✘ Errors were logged during apply operation. Please check component installation logs above.\n")
 		return fmt.Errorf("errors were logged during apply operation")
 	}
-	l.logAndPrint("\n\n✔ Installation complete\n")
+	if wait {
+		objs, err := object.ParseK8sObjectsFromYAMLManifest(reconciler.GetManifests().String())
+		if err != nil {
+			l.LogAndPrintf("\n\n✘ Errors in manifest:\n%s\n", err)
+			return fmt.Errorf("errors during wait")
+		}
+		if err := manifest.WaitForResources(objs, clientSet, waitTimeout, dryRun); err != nil {
+			l.LogAndPrintf("\n\n✘ Errors during wait:\n%s\n", err)
+			return fmt.Errorf("errors during wait")
+		}
+	}
 
-	crName := installedSpecCRPrefix
-	if iops.Revision != "" {
-		crName += "-" + iops.Revision
+	l.LogAndPrint("\n\n✔ Installation complete\n")
```

Comment:
yes, the error handling is still wip because the controller has apply loops per component. i need to serialize the return codes through a channel and wait for all to complete as well as figuring out the some feedback to the user.